numpy/core/code_generators/generate_umath.py

```diff
@@ -664,7 +664,7 @@ def english_upper(s):
           None,
           TD('e', f='cos', astype={'e':'f'}),
           TD('f', simd=[('fma', 'f'), ('avx512f', 'f')]),
-          TD(inexact, f='cos', astype={'e':'f'}),
+          TD('fdg' + cmplx, f='cos'),
```

Comment:
Are you sure this works with/without the simd variants now?

Reply:
The simd variant are only generated for type f.  This change removes the duplicated signature ee->e.